hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -368,58 +363,69 @@ func getProperties(
 	}
 
 	// see: https://json-schema.org/understanding-json-schema/reference/object.html#properties
-	if schema.AdditionalProperties == nil {
-		// if not specified, any additional properties are allowed (TODO: tell all Azure teams this fact and get them to update their API definitions)
-		// for now we aren't following the spec 100% as it pollutes the generated code
-		// only generate this property if there are no other properties:
-		if len(properties) == 0 {
+	if schema.additionalPropertiesAllowed() {
+		additionalPropSchema := schema.additionalPropertiesSchema()
+		if additionalPropSchema == nil {
+			// if not specified, any additional properties are allowed (TODO: tell all Azure teams this fact and get them to update their API definitions)
```

Comment:
extra minor: We're linebreaking here slightly oddly -- first line is really long and the subsequent ones aren't.

IMO, break the first line either in the parenthesis statement, or maybe just break before it starts and put that whole statement onto it's own line?